src/components/portrait/portrait.pw.tsx

```diff
@@ -0,0 +1,486 @@
+
+      await portraitPreview(page).hover();
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      await expect(tooltip).toBeVisible();
+      await expect(tooltip).toHaveText(tooltipMessage);
+    });
+  });
+
+  testData.forEach((tooltipId) => {
+    test(`should render with tooltipId as ${tooltipId}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <PortraitComponent tooltipMessage="foo" tooltipId={tooltipId} />
+      );
+
+      await portraitPreview(page).hover();
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      await expect(tooltip).toBeVisible();
+      await expect(tooltip).toHaveId(tooltipId);
+    });
+  });
+
+  visibilityValues.forEach(([boolVal, renderState]) => {
+    test(`should render ${renderState} a tooltip, when visibility prop is ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <PortraitComponent tooltipMessage="foo" tooltipIsVisible={boolVal} />
+      );
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      if (boolVal) {
+        await expect(tooltip).toBeVisible();
+        await expect(tooltip).toHaveText("foo");
+      } else {
+        await expect(tooltip).not.toBeVisible();
+      }
+    });
+  });
+
+  ["top", "bottom", "left", "right"].forEach((tooltipPosition) => {
+    test(`should render with tooltip positioned ${tooltipPosition}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <Box p={200}>
+          <PortraitComponent tooltipPosition={tooltipPosition} />{" "}
+        </Box>
+      );
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      await expect(tooltip).toBeVisible();
+      await expect(tooltip).toHaveAttribute("data-placement", tooltipPosition);
+    });
+  });
+
+  test("should render with a tooltip error", async ({ mount, page }) => {
+    await mount(<PortraitComponent tooltipType="error" />);
+
+    const tooltip = getDataElementByValue(page, "tooltip");
+
+    await expect(tooltip).toHaveCSS("background-color", VALIDATION.ERROR);
+  });
+
+  [
+    [SIZE.MEDIUM, "14px"],
+    [SIZE.LARGE, "16px"],
+  ].forEach(([tooltipSize, fontSize]) => {
+    test(`should render with a ${tooltipSize} size tooltip`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PortraitComponent tooltipSize={tooltipSize} />);
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      await expect(tooltip).toBeVisible();
+      await expect(tooltip).toHaveCSS("font-size", fontSize);
+    });
+  });
+
+  colors.forEach(([names, color]) => {
+    test(`should render with a tooltip with a ${names} background color`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PortraitComponent tooltipBgColor={color} />);
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+
+      await expect(tooltip).toBeVisible();
```

Comment:
**question** (non-blocking): I notice this visibility check wasn't in the equivalent Cypress tests, so while I don't object particularly, I'm wondering why you've introduced it here?

I'm actually a bit concerned because I thought the tooltip shouldn't be visible till you focus/hover it (and it shouldn't be possible to check the background color until it is!), yet this apparently all works here without doing that! Kind of makes me wonder what the value was in those tests further up where we explicitly hover before asserting the tooltip is visible.

Mind you I don't know the component very well, and if these are problems then they existed before your Playwright refactor 😄 Mostly I'm wondering why we need the visibility check here - the rest can be investigated and tickets raised if something needs fixing.

Reply:
Yeah, looks like these were just added to the colour related tests in error, I'll do some further tests on tests which require visibility tests and `.hover()` just to make sure there aren't any issues and the tests aren't redundant

Reply:
Turns out this passed because the pre-made `Portrait` component within `components.test-pw.stx` had `tooltipVisibility` set to true. Turns out this component was also used in other tests where we're using focus and checking visibility, so I'll change that